lint/analyzers_test.go

```diff
@@ -42,6 +44,18 @@ func testAnalyzers(t *testing.T, code string, analyzers ...*analysis.Analyzer) [
 		nil,
 	)
 
+	if suppressParsingCheckingErrors {
+		// Suppress parser errors
+		config.HandleParserError = func(err analysis.ParsingCheckingError, program *ast.Program) error {
+			return nil
+		}
+
+		// Suppress checker errors
+		config.HandleCheckerError = func(err analysis.ParsingCheckingError, checker *sema.Checker) error {
+			return nil
+		}
```

Comment:
Conceptual question: Why are these handlers set in tests, but not during "normal" operation? Are they going to be set outside of the linter, e.g. in the LS?

Reply:
Yeah, the LS effectively does this itself already (i.e. it parses & checks the program itself, reports errors as diagnostics and treats as non-fatal, and instantiates an `analysis.Program` itself as oppose to using `analysis.Load`).  Ideally it would probably just use `analysis.Load` instead of rewriting this logic itself.

I'm hoping to use this functionality when adding the linter to the Flow CLI so we can surface the Cadence 1.0 diagnostics - previously `analysis.Load` would have always aborted once it encountered parser/checker errors.  (i.e. we can use `analysis.Load(...)` => `programs[xxx].Run(LintingAnalyzers)` instead of having to rewrite this code in a 3rd place).

In terms of the "normal" functionalities exported by `lint.go`, I haven't touched these at all because these seemed somewhat tailored towards use-cases which I am not very familiar with and was unsure what expected behaviour was (i.e. would recovering from these checker/parser errors be undesirable or unnecessary).  If we see value in capturing & recovering from these errors in this code I can change it as well.